src/css/components/modal.less

```diff
@@ -109,3 +119,99 @@
     }
   }
 }
+
+.modal.dialog {
+  color: @modal-dialog-text-color;
+  font-size: @modal-font-size;
+
+  .modal-header {
+    background-color: transparent;
+    color: @modal-dialog-info-color;
+    font-size: @modal-font-size-header;
+    padding-top: @modal-dialog-padding;
+    padding-bottom: 0;
+    text-align: center;
+  }
+
+  .modal-content {
+    background-color: @modal-dialog-bg-color;
+    border: none;
+    box-shadow: none;
+    min-width: @modal-dialog-width;
+
+    &::before {
+      content: " ";
+      position: absolute;
+      top: 0;
+      left: 0;
+      width: 100%;
+      height: @base-spacing-unit;
+      background-color: @modal-dialog-info-color;
+      border-radius: @modal-border-radius @modal-border-radius 0 0;
+    }
+  }
+
+  .modal-body {
+    text-align: center;
+    padding-bottom: @modal-dialog-padding;
+
+    label {
+      font-weight: inherit;
+    }
+
+    .form-control {
+      width: auto;
+      margin-top:  @modal-body-padding;
+      margin-left:  auto;
+      margin-right:  auto;
+    }
+  }
+
+  .modal-footer {
+    background-color: @modal-dialog-footer-color;
+  }
+
+  &.warning {
```

Comment:
Just throwing my 2 cents here: I think we try to always put class-level modifiers before children when nesting – it's more obvious what all the possible "states" are.